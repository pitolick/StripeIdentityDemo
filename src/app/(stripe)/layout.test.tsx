import React from 'react'
import { render } from '@testing-library/react'
import FormStripeLayout from './layout'

// StripeContextProvider, MicroCMSContextProviderをモック化する
jest.mock('@/features/context', () => ({
  StripeContextProvider: ({ children }: { children: React.ReactNode }) => <div data-testid="StripeContextProvider">{children}</div>,
}))

describe('FormStripeLayout', () => {
  it('子コンポーネントを正しくレンダリングすること', () => {
    const { container } = render(
      <FormStripeLayout>
        <div>子コンポーネント</div>
      </FormStripeLayout>,
    )

    expect(container).toMatchSnapshot()
  })
})
