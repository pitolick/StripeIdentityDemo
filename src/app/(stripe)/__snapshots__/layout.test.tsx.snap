// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormStripeLayout 子コンポーネントを正しくレンダリングすること 1`] = `
<div>
  <div
    data-testid="StripeContextProvider"
  >
    <div>
      子コンポーネント
    </div>
  </div>
</div>
`;
